src/shogun/optimization/WrappedMinimizer.cpp

```diff
@@ -0,0 +1,205 @@
+	:CFirstOrderMinimizer(fun)
+{
+	init();
+}
+
+void CWrappedMinimizer::init()
+{
+	m_variable_vec=SGVector<float64_t>();
+	m_variable=NULL;
+	m_is_in_place=false;
+
+	SG_ADD(&m_variable_vec,"variable_vec","variable_vec", MS_NOT_AVAILABLE);
+	SG_ADD(&m_is_in_place,"is_in_place","is_in_place", MS_NOT_AVAILABLE);
+	SG_ADD((CSGObject **)&m_variable,"m_variable","m_variable", MS_NOT_AVAILABLE);
+}
+
+SGVector<float64_t> CWrappedMinimizer::convert(CMap<TParameter*, SGVector<float64_t> >* input)
+{
+	REQUIRE(input,"Input not set\n");
+
+	int32_t length=0;
+	for(int32_t i=0; i<input->get_num_elements(); i++)
+	{
+		CMapNode<TParameter*, SGVector<float64_t> >* var=input->get_node_ptr(i);
+		REQUIRE(var->key->m_name, "The name of variable not set \n");
+		REQUIRE((var->data).vlen>0,
+			"The length (%d) of variable (%s) is empty \n",
+			(var->data).vlen, var->key->m_name);
+		length+=(var->data).vlen;
+	}
+	SGVector<float64_t> output (length);
+	int32_t offset=0;
+	if (length>0)
+	{
+		for(int32_t i=0; i<input->get_num_elements(); i++)
+		{
+			CMapNode<TParameter*, SGVector<float64_t> >* var=input->get_node_ptr(i);
+			int32_t vlen=(var->data).vlen;
+			std::copy((var->data).vector, (var->data).vector+vlen, output.vector+offset);
+			offset+=vlen;
+		}
+	}
+	ASSERT(offset==length);
+	return output;
+}
+void CWrappedMinimizer::copy_in_parameter_order(const float64_t* input, const int32_t dim,
+	CMap<TParameter*, SGVector<float64_t> >* parameter_order,
+	CMap<TParameter*, SGVector<float64_t> >* output)
+{
+	REQUIRE(input,"Input not set\n");
+	REQUIRE(output,"Output not set\n");
+	REQUIRE(parameter_order,"Parameter_order not set\n");
+	REQUIRE(dim>0,"The length of input (%d) is not positive\n",dim);
+
+	REQUIRE(output->get_num_elements()==parameter_order->get_num_elements(),
+		"The length of output (%d) and the length of parameter_order (%d) do not match\n",
+		output->get_num_elements(),parameter_order->get_num_elements());
+
+	int32_t offset=0;
+	for(int32_t i=0; i<parameter_order->get_num_elements(); i++)
+	{
+		CMapNode<TParameter*, SGVector<float64_t> >* var=parameter_order->get_node_ptr(i);
+
+		int32_t j=0;
+		for(; j<output->get_num_elements(); j++)
+		{
+			CMapNode<TParameter*, SGVector<float64_t> >* g=output->get_node_ptr(j);
+			REQUIRE(g->key->m_name, "The name of parameter not set \n");
+			if (strcmp(var->key->m_name,g->key->m_name)==0)
+			{
+				int32_t vlen=(g->data).vlen;
+				REQUIRE((var->data).vlen==(g->data).vlen,
+					"The dim (%d) of parameter %s and the length of its output (%d) do not match\n",
+					(var->data).vlen,var->key->m_name,(g->data).vlen);
+				REQUIRE(offset+vlen<=dim,"")
+				std::copy(input+offset, input+offset+vlen,(g->data).vector);
+				offset+=vlen;
+				break;
+			}
+		}
+		REQUIRE(j<output->get_num_elements(), "Cannot find %s in output \n",
+			var->key->m_name);
+	}
+	ASSERT(offset==dim);
+}
+
+void CWrappedMinimizer::copy_in_parameter_order(CMap<TParameter*, SGVector<float64_t> >* input,
```

Comment:
Since the assumption above depends on the type/class of the `data` field, I can write some std helper functions for specific classes (for example, SGVector<float64_t>) in the `data` field.